File: core/src/main/java/javabot/operations/RegisterNickOperation.java
package javabot.operations;

import com.antwerkz.sofia.Sofia;
import javabot.Message;
import javabot.dao.AdminDao;
import javabot.dao.ConfigDao;
import javabot.model.Config;
import javabot.model.NickRegistration;

import javax.inject.Inject;

public class RegisterNickOperation extends BotOperation {
    @Inject
    private AdminDao adminDao;
    @Inject
    private ConfigDao configDao;

    @Override
    public boolean handleMessage(Message event) {
        final String message = event.getValue();
        if (message.startsWith("register ")) {
            String[] split = message.split(" ");
            if (split.length > 1) {
                String twitterName = split[1];
                NickRegistration registration = new NickRegistration(event.getUser(), twitterName);
                adminDao.save(registration);
                Config config = configDao.get();
                String eventMessage = Sofia.registerNick(config.getUrl(), registration.getUrl(), twitterName);

                getBot().postMessage(null, event.getUser(), eventMessage, event.isTell());
                return true;
            }
        }
        return false;
    }
}

File: src/main/kotlin/javabot/operations/RegisterNickOperation.kt
package javabot.operations

import com.antwerkz.sofia.Sofia
import jakarta.inject.Inject
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javabot.dao.ConfigDao
import javabot.model.NickRegistration

class RegisterNickOperation
@Inject
constructor(bot: Javabot, adminDao: AdminDao, var configDao: ConfigDao) :
    BotOperation(bot, adminDao) {
    override fun handleMessage(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val message = event.value
        if (message.startsWith("register ")) {
            val split = message.split(" ")
            if (split.size > 1) {
                val twitterName = split[1]
                val registration = NickRegistration(event.user, twitterName)
                adminDao.save(registration)

                responses.add(
                    Message(
                        event.user,
                        Sofia.registerNick(configDao.get().url, registration.url ?: "", twitterName)
                    )
                )
            }
        }
        return responses
    }
}
